extensions/project-descriptor.md

```diff
@@ -188,6 +190,19 @@ Either a `version`, `uri`, or `script` table MUST be included, but MUST NOT incl
 
 The `api` and `inline` key MUST be defined in the `script` table. The value of the `inline` key will be used as the build script for the [inline buildpack](#Definitions) this entry represents. The value of the `api` key defines its Buildpack API compatibility, and the `shell` key defines the shell used to execute the `inline` script.
 
+#### `io.buildpacks.pre.group` (optional)
+
+This table MAY contain a list of buildpacks to insert at the beginning of an automatically detected group. Given an order with multiple groups, the list of `pre` buildpacks will be inserted at the beginning of each group such that they are run as if they were original included in the group. Each phase of the injected buildpack(s) will execute as normal.
```

Comment:
```suggestion
This table MAY contain a list of buildpacks to insert at the beginning of an automatically detected group. Given an order with multiple groups, the list of `pre` buildpacks will be inserted at the beginning of each group such that they are run as if they were originally included in the group. Each phase of the injected buildpack(s) will execute as normal.
```

Reply:
The RFC states:
> injected into the beginning or end of the groups defined in an order.toml provided by a builder.

I initially read `insert at the beginning of an automatically detected group` as a post-detection step. But I think it is meant to say something like `insert at the beginning of all defined groups`.

Reply:
Another question regarding insertion of these buildpacks. Do these insertions happen to the group defined _inside_ of a project descriptor? Or do these only get injected for builder's `order.toml`?

Reply:
@jabrown85 

>But I think it is meant to say something like insert at the beginning of all defined groups.

oh, that's a subtle but important different. I think it means detect will be run. I'll update this PR to use that langauge.

> Do these insertions happen to the group defined inside of a project descriptor? Or do these only get injected for builder's order.toml?

Yes, i'll add that.

Reply:
Actually, i take that last part back. There shouldn't be a need to use `pre/`post` if you're defining a group in project.toml. I'm not sure what the behavior should be if both are used though.

Reply:
> Actually, i take that last part back. There shouldn't be a need to use `pre/`post` if you're defining a group in project.toml. I'm not sure what the behavior should be if both are used though.

maybe it should be an "or", where you can't use `pre`/`post` but not with a normal buildpack group.

Reply:
From the leadership WG meeting, what happens on the `pack` CLI when overriding with `-b`? Will builders support `pre`/`post` with considerations like buildpacks/rfcs#175?